script/nan-spec-runner.js

```diff
@@ -25,7 +25,43 @@ async function main () {
     npm_config_arch: process.env.NPM_CONFIG_ARCH,
     npm_config_yes: 'true'
   });
-  const { status: buildStatus } = cp.spawnSync(NPX_CMD, ['node-gyp', 'rebuild', '--directory', 'test', '-j', 'max'], {
+
+  const clangDir = path.resolve(BASE, 'third_party', 'llvm-build', 'Release+Asserts', 'bin');
+  const cc = path.resolve(clangDir, 'clang');
+  const cxx = path.resolve(clangDir, 'clang++');
+  const ld = path.resolve(clangDir, 'lld');
+
+  // TODO(ckerr) this is cribbed from read obj/electron/electron_app.ninja.
+  // Maybe it would be better to have this script literally open up that
+  // file and pull cflags_cc from it instead of using bespoke code here?
+  // I think it's unlikely to work; but if it does, it would be more futureproof
+  const cxxflags = [
+    '-std=c++14',
+    '-nostdinc++',
+    '-D_LIBCPP_HAS_NO_VENDOR_AVAILABILITY_ANNOTATIONS', // needed by next line
+    `-isystem"${path.resolve(BASE, 'buildtools', 'third_party', 'libc++', 'trunk', 'include')}"`,
+    `-isystem"${path.resolve(BASE, 'buildtools', 'third_party', 'libc++abi', 'trunk', 'include')}"`,
+    '-fPIC'
```

Comment:
hm, isn't `-fPIC` already passed by other code (e.g. gyp)?

Reply:
It is on linux x64, but it's not passed by default for linux IA32. This flag was added to pass on IA32, I can look into a better solution

Reply:
ah i see: https://github.com/nodejs/node-gyp/blob/4d036526c5347ad6fb0394993bda05f8260f7ae4/addon.gypi#L180-L181

that's a bit surprising. i wonder why this isn't passed on ia32? it looks like it's been that way forever. not that we care terribly much about ia32, but does it break things to pass -fPIC there?

Reply:
@nornagon It probably doesn't break anything for us, but we can't guarantee that for node and all of it's native modules.  Like it clearly works in our ia32 CI but also that isn't technically running on ia32 hardware either 🤷  Electron has shipped built with PIC since like 2017 so if there are issues with PIC specific to 32 bit linux native modules that would be super weird 😆